### ISSUE TITLE
dns_not_found_ttl and dns_error_ttl defaults are swapped

### ISSUE THREAD

**Issue by chris-branch** [Kong/kong#5675]
## Summary

The documentation descriptions and default values are out of sync with the code for several of the DNS configuration options, and as a result I believe Kong's defaults are incorrect (or at least not what was intended):

### Documentation:

`dns_not_found_ttl`
TTL in seconds for empty DNS responses and “(3) name error” responses.
Default: `30`

`dns_error_ttl`
TTL in seconds for error responses.
Default: `1`

### Kong Default Config:
https://github.com/Kong/kong/blob/master/kong/templates/kong_defaults.lua#L109

`dns_not_found_ttl = 30`
`dns_error_ttl = 1`

### Kong Code:
https://github.com/Kong/kong/blob/master/kong/tools/dns.lua#L31

```
badTtl = conf.dns_not_found_ttl, -- ttl in seconds for dns error responses (except 3 - name error)
emptyTtl = conf.dns_error_ttl,   -- ttl in seconds for empty and "(3) name error" dns responses
```

### DNS Client Code:
https://github.com/Kong/lua-resty-dns-client/blob/master/src/resty/dns/client.lua#L408

```
-- local emptyTtl = 30.0   -- in seconds (can have fractions)
-- local badTtl = 1.0      -- in seconds (can have fractions)
```

The net result is that the default values that end up in a running instance of Kong are:

`emptyTtl = 1.0`
`badTtl = 30.0`

which is backwards from what is documented.  So, a missing name or empty response will re-query every second, while other errors will wait 30 seconds before re-query.

## Steps To Reproduce

1. Start Kong with all default options for DNS, and LOG_LEVEL=debug
2. Review config output that appears in the log (example below)

## Additional Details & Logs

- Kong version: 1.4.2

Partial log output at startup with LOG_LEVEL=debug and all default settings for DNS:

```
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:447: init(): [dns-client] (re)configuring dns client
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:452: init(): [dns-client] staleTtl = 4
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:455: init(): [dns-client] validTtl = nil
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:459: init(): [dns-client] noSynchronisation = false
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:478: init(): [dns-client] query order = LAST, SRV, A, CNAME
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:577: init(): [dns-client] nameserver 127.0.0.11
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:582: init(): [dns-client] attempts = 5
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:591: init(): [dns-client] timeout = 2000 ms
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:595: init(): [dns-client] ndots = 0
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:597: init(): [dns-client] search = 
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:603: init(): [dns-client] badTtl = 30 s
2020/03/12 16:11:50 [debug] 1#0: [lua] client.lua:605: init(): [dns-client] emptyTtl = 1 s
```

Note the values for `badTtl` and `emptyTtl`

**Comment by hishamhm** [comment]
@chris-branch Thank you for the detailed report! @Tieske, is this just a matter of swapping `badTtl` and `emptyTtl` in `kong/tools/dns.lua`?

**Comment by chris-branch** [comment]
My opinion: Yes, swapping the variable assignments in `kong/tools/dns.lua` seems like the right fix.  But I would also suggest adding a warning to the changelog/readme for anyone who is upgrading and has set non-defaults for either config value because this code change would effectively end up swapping their custom values.

**Comment by Tieske** [comment]
@hishamhm yes, that should fix it, see #5684

**Comment by hishamhm** [comment]
@Tieske thanks for the PR! Merged!

@chris-branch Thank you for the report! It will be correct in the next release!
